cmd/pinniped-proxy/src/https.rs

```diff
@@ -71,18 +71,12 @@ pub fn get_api_server_url(request_headers: &HeaderMap<HeaderValue>) -> Result<St
 }
 
 /// get_api_server_cert_auth_data returns a byte vector result containing the base64 decoded value.
-pub fn get_api_server_cert_auth_data(request_headers: &HeaderMap<HeaderValue>) -> Result<Vec<u8>> {
-    match request_headers.get(HEADER_K8S_API_SERVER_CA_CERT) {
-        Some(header_value_b64) => match base64::decode(header_value_b64.as_bytes()) {
-            Ok(data) => Ok(data),
-            Err(e) => {
-                debug!("failed to base64 decode {} header data: {}", HEADER_K8S_API_SERVER_CA_CERT, e);
-                Err(anyhow::anyhow!(e))
-            }
-        },
-        None => {
-            debug!("header {} required but not present", HEADER_K8S_API_SERVER_CA_CERT);
-            Err(anyhow::anyhow!("header {} required but not present", HEADER_K8S_API_SERVER_CA_CERT))
+pub fn get_api_server_cert_auth_data(cacert_header: &HeaderValue) -> Result<Vec<u8>> {
```

Comment:
Just updating this function to take the actual header value as an arg instead of the headers, which removes the need to error if it's not present in the headers (moved that out to the service).